src/tests/JIT/opt/ObjectStackAllocation/ObjectStackAllocationTests.cs

```diff
@@ -174,6 +179,8 @@ static bool SPCOptimizationsEnabled()
             Assembly objectAssembly = Assembly.GetAssembly(typeof(object));
             object[] attribs = objectAssembly.GetCustomAttributes(typeof(DebuggableAttribute),
                                                         false);
+            if (attribs.Length == 0)
+                return true; // Assume corelib is optimized in this case
```

Comment:
@MichalStrehovsky do we run runtime tests with NativeAOT in full debug mode?

Reply:
So can you tell me how to properly check it here?

Reply:
> @MichalStrehovsky do we run runtime tests with NativeAOT in full debug mode?

We do have such legs, I'm not sure if they include this test too. It would be in the runtime or runtime-nativeaot-outerloop if it exists.

> So can you tell me how to properly check it here?

Native AOT will use the same settings for optimizing both CoreLib and user code so enabling the attribute and looking for the attribute on `Assembly.GetEntryAssembly` would give the answer.

I guess I don't understand why this is looking at whether corelib is optimized. Why does that affect stack allocations in this test?

Reply:
> I guess I don't understand why this is looking at whether corelib is optimized. Why does that affect stack allocations in this test?

Maybe the way we build the tests, optimization of CoreLib always matches optimization of the test?

If that's the case, all that's needed is to add `<DebuggerSupport>true</DebuggerSupport>` to the test project and this will work the same. It will also fail the same if there's ever mismatch between test optimization and corelib optimization.

Reply:
> I guess I don't understand why this is looking at whether corelib is optimized. Why does that affect stack allocations in this test?

I am not the original author (I presume the authors no longer work in the team), but looks like the test tries to be agile and validate that nothing is stack-allocated under debug maybe? 🤷

Reply:
(I mean in the JIT case. It would also fail in the AOT case because I believe the test is looking at the wrong assembly.)

Reply:
I ended up removing these bits and the test still working fine (it's always optimized due to `<Optimize>true</Optimize>`)